src/template_utils.rs

```diff
@@ -63,7 +63,7 @@ impl<'r> Responder<'r> for Ructe {
         let etag = format!("{:x}", hasher.finish());
         if r.headers()
             .get("If-None-Match")
-            .any(|s| s[1..s.len() - 1] == etag)
+            .any(|s| s[1..s.len() - 1] == etag || s[3..s.len() - 1] == etag)
```

Comment:
can one of you two explain what this does?

Reply:
this should probably get it's comment actually.
We send Etags as `'some_tag'`, the old match does not verify the `'`, but check that whatever in-between is the right tag. Nginx apparently change the Etag to `W/'some_tag'`, W/ indicating it's a weak Etag, the content may not be byte-equal, but is semantically identical (which is imo a shitty but correct interpretation of the RFC, as Nginx send compressed data)

The new match ignore that W/ saying it's a weak etag and so match if Nginx modified the header we sent previously

Comment might looks like : // if a strong or weak etag matches